app/src/test/java/protect/card_locker/LoyaltyCardViewActivityTest.java

```diff
@@ -405,7 +405,9 @@ public void noDataLossOnResumeOrRotate() {
             storeField.setText("correct store");
             noteField.setText("correct note");
             LoyaltyCardEditActivity.formatDateField(context, validFromField, validFromDate);
+            activity.updateTempState(LoyaltyCardField.validFrom, validFromDate);
             LoyaltyCardEditActivity.formatDateField(context, expiryField, expiryDate);
+            activity.updateTempState(LoyaltyCardField.expiry, expiryDate);
```

Comment:
I'm not a huge fan of copying these specific calls into tests as it makes the tests untrustworthy (they will pass even when the code is changed to break it again). But I will still merge it if this fixes it because the test is not **more untrustworthy** than before at least (still have to test your patch)

Reply:
Yeah, I wasn't happy with that either. But it was needed. I can wrap both calls in a new function instead if you prefer.

Reply:
I tested with Android 13 (emulator). But more testing is always a good idea.

Reply:
Not sure the test should be calling `formatDateField()` directly either. But I didn't want to mess with things more than necessary for this PR. Happy to follow up with some cleanup if you want.

Reply:
> Not sure the test should be calling formatDateField() directly either. But I didn't want to mess with things more than necessary for this PR. Happy to follow up with some cleanup if you want.

Yeah, that's what I meant with the test is not "more untrustworthy" because it already was untrustworthy.

A follow-up at some point would be great, but out of scope. Just wanted to make sure we agreed here :)

Reply:
Yeah, definitely :)